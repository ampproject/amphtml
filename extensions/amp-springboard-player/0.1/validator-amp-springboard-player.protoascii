#
# Copyright 2016 The AMP HTML Authors. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS-IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the license.
#

tags: {  # amp-springboard-player
  html_format: AMP
  html_format: AMP4ADS
  tag_name: "SCRIPT"
  spec_name: "amp-springboard-player extension .js script"
  mandatory_parent: "HEAD"
  unique_warning: true
  extension_unused_unless_tag_present: "amp-springboard-player"
  attrs: {
    name: "async"
    mandatory: true
    value: ""
  }
  attrs: {
    name: "custom-element"
    mandatory: true
    value: "amp-springboard-player"
    dispatch_key: true
  }
  attrs: { name: "nonce" }
  attrs: {
    name: "src"
    mandatory: true
    value_regex: "https://cdn\\.ampproject\\.org/v0/amp-springboard-player-(latest|0\\.1)\\.js"
  }
  attrs: {
    name: "type"
    value_casei: "text/javascript"
  }
  cdata: {
    blacklisted_cdata_regex: {
      regex: "."
      error_message: "contents"
    }
  }
  spec_url: "https://www.ampproject.org/docs/reference/extended/amp-springboard-player.html"
}
tags: {  # <amp-springboard-player>
  html_format: AMP
  html_format: AMP4ADS
  tag_name: "AMP-SPRINGBOARD-PLAYER"
  disallowed_ancestor: "AMP-SIDEBAR"
  also_requires_tag: "amp-springboard-player extension .js script"
  attrs: {
    name: "data-content-id"
    mandatory: true
  }
  attrs: {
    name: "data-domain"
    mandatory: true
  }
  attrs: {
    name: "data-items"
    mandatory: true
  }
  attrs: {
    name: "data-mode"
    mandatory: true
    value_regex_casei: "playlist|video"
  }
  attrs: {
    name: "data-player-id"
    mandatory: true
    value_regex_casei: "[a-z0-9]+"
  }
  attrs: {
    name: "data-site-id"
    mandatory: true
    value_regex: "[0-9]+"
  }
  attr_lists: "extended-amp-global"
  spec_url: "https://www.ampproject.org/docs/reference/extended/amp-springboard-player.html"
  amp_layout: {
    supported_layouts: FILL
    supported_layouts: FIXED
    supported_layouts: FLEX_ITEM
    supported_layouts: RESPONSIVE
  }
}
