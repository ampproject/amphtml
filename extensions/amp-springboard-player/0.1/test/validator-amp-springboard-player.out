FAIL
|  <!--
|    Copyright 2015 The AMP HTML Authors. All Rights Reserved.
|
|    Licensed under the Apache License, Version 2.0 (the "License");
|    you may not use this file except in compliance with the License.
|    You may obtain a copy of the License at
|
|        http://www.apache.org/licenses/LICENSE-2.0
|
|    Unless required by applicable law or agreed to in writing, software
|    distributed under the License is distributed on an "AS-IS" BASIS,
|    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
|    See the License for the specific language governing permissions and
|    limitations under the license.
|  -->
|  <!--
|    Test Description:
|    This tests validation for the amp-springboard-player tag.
|  -->
|  <!doctype html>
|  <html âš¡>
|  <head>
|      <meta charset="utf-8">
|      <link rel="canonical" href="amps.html">
|      <meta name="viewport" content="width=device-width,minimum-scale=1,initial-scale=1">
|      <link href="https://fonts.googleapis.com/css?family=Questrial" rel="stylesheet" type="text/css">
|      <script async custom-element="amp-springboard-player" src="https://cdn.ampproject.org/v0/amp-springboard-player-0.1.js"></script>
|       <style amp-boilerplate>body{-webkit-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-moz-animation:-amp-start 8s steps(1,end) 0s 1 normal both;-ms-animation:-amp-start 8s steps(1,end) 0s 1 normal both;animation:-amp-start 8s steps(1,end) 0s 1 normal both}@-webkit-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-moz-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-ms-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@-o-keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}@keyframes -amp-start{from{visibility:hidden}to{visibility:visible}}</style><noscript><style amp-boilerplate>body{-webkit-animation:none;-moz-animation:none;-ms-animation:none;animation:none}</style></noscript>
|      <script async src="https://cdn.ampproject.org/v0.js"></script>
|  </head>
|  <body>
|    <!-- valid -->
|    <amp-springboard-player data-site-id="261" data-mode="video"
|        data-content-id="1578473" data-player-id="test401"
|        data-domain="test.com" data-items="10"
|        layout="fixed" width="480" height="270"></amp-springboard-player>
|    <!-- invalid: data-site-id must be numeric -->
|    <amp-springboard-player data-site-id="a261" data-mode="video"
>>   ^~~~~~~~~
amp-springboard-player/0.1/test/validator-amp-springboard-player.html:38:2 The attribute 'data-site-id' in tag 'amp-springboard-player' is set to the invalid value 'a261'. (see https://amp.dev/documentation/components/amp-springboard-player)
|        data-content-id="1578473" data-player-id="test401"
|        data-domain="test.com" data-items="10"
|        layout="fixed"width="480" height="270"></amp-springboard-player>
|    <!-- invalid: data-player-id must be alphanumeric -->
|    <amp-springboard-player data-site-id="261" data-mode="playlist"
>>   ^~~~~~~~~
amp-springboard-player/0.1/test/validator-amp-springboard-player.html:43:2 The attribute 'data-player-id' in tag 'amp-springboard-player' is set to the invalid value 'test 401'. (see https://amp.dev/documentation/components/amp-springboard-player)
|        data-content-id="1578473" data-player-id="test 401"
|        data-domain="test.com" data-items="10"
|        layout="fixed" width="480" height="270"></amp-springboard-player>
|    <!-- invalid: data-content-id required -->
|    <amp-springboard-player data-site-id="261" data-mode="video"
>>   ^~~~~~~~~
amp-springboard-player/0.1/test/validator-amp-springboard-player.html:48:2 The mandatory attribute 'data-content-id' is missing in tag 'amp-springboard-player'. (see https://amp.dev/documentation/components/amp-springboard-player)
|        data-player-id="test401"
|        data-domain="test.com" data-items="10"
|        layout="fixed" width="480" height="270"></amp-springboard-player>
|    <!-- invalid: data-domain required -->
|    <amp-springboard-player data-site-id="261" data-mode="video"
>>   ^~~~~~~~~
amp-springboard-player/0.1/test/validator-amp-springboard-player.html:53:2 The mandatory attribute 'data-domain' is missing in tag 'amp-springboard-player'. (see https://amp.dev/documentation/components/amp-springboard-player)
|        data-content-id="1578473" data-player-id="test401"
|        data-items="10"
|        layout="fixed" width="480" height="270"></amp-springboard-player>
|    <!-- invalid: data-items required -->
|    <amp-springboard-player data-site-id="261" data-mode="video"
>>   ^~~~~~~~~
amp-springboard-player/0.1/test/validator-amp-springboard-player.html:58:2 The mandatory attribute 'data-items' is missing in tag 'amp-springboard-player'. (see https://amp.dev/documentation/components/amp-springboard-player)
|        data-content-id="1578473" data-player-id="test401"
|        data-domain="test.com"
|        layout="fixed" width="480" height="270"></amp-springboard-player>
|    <!-- invalid: data-mode required -->
|    <amp-springboard-player data-site-id="261"
>>   ^~~~~~~~~
amp-springboard-player/0.1/test/validator-amp-springboard-player.html:63:2 The mandatory attribute 'data-mode' is missing in tag 'amp-springboard-player'. (see https://amp.dev/documentation/components/amp-springboard-player)
|        data-content-id="1578473" data-player-id="test401"
|        data-domain="test.com" data-items="10"
|        layout="fixed" width="480" height="270"></amp-springboard-player>
|    <!-- invalid: data-player-id required -->
|    <amp-springboard-player data-site-id="261" data-mode="video"
>>   ^~~~~~~~~
amp-springboard-player/0.1/test/validator-amp-springboard-player.html:68:2 The mandatory attribute 'data-player-id' is missing in tag 'amp-springboard-player'. (see https://amp.dev/documentation/components/amp-springboard-player)
|        data-content-id="1578473"
|        data-domain="test.com" data-items="10"
|        layout="fixed" width="480" height="270"></amp-springboard-player>
|    <!-- invalid: data-site-id required -->
|    <amp-springboard-player data-mode="video"
>>   ^~~~~~~~~
amp-springboard-player/0.1/test/validator-amp-springboard-player.html:73:2 The mandatory attribute 'data-site-id' is missing in tag 'amp-springboard-player'. (see https://amp.dev/documentation/components/amp-springboard-player)
|        data-content-id="1578473" data-player-id="test401"
|        data-domain="test.com" data-items="10"
|        layout="fixed" width="480" height="270"></amp-springboard-player>
|    <!-- invalid: bad layout -->
|    <amp-springboard-player data-site-id="261" data-mode="playlist"
>>   ^~~~~~~~~
amp-springboard-player/0.1/test/validator-amp-springboard-player.html:78:2 The specified layout 'FIXED_HEIGHT' is not supported by tag 'amp-springboard-player'. (see https://amp.dev/documentation/components/amp-springboard-player)
|        data-player-id="test401"
|        layout="fixed-height" width="480" height="270"></amp-springboard-player>
|  </body>
|  </html>
